iree/compiler/InputConversion/MHLO/ConvertMHLOToLinalgExt.cpp

```diff
@@ -161,25 +167,55 @@ struct SortOpConversion : public OpConversionPattern<mhlo::SortOp> {
   using OpConversionPattern<mhlo::SortOp>::OpConversionPattern;
 
   LogicalResult matchAndRewrite(
-      mhlo::SortOp mhloSortOp, OpAdaptor adaptor,
+      mhlo::SortOp op, OpAdaptor adaptor,
       ConversionPatternRewriter &rewriter) const final {
+    Location loc = op.getLoc();
+    llvm::SmallVector<Value> inputs;
+    for (auto input : adaptor.getOperands()) {
+      Type newType = convertShapedToSignless(input.getType());
+      if (newType == input.getType()) {
+        inputs.push_back(input);
+      } else {
+        inputs.push_back(
+            rewriter.create<UnrealizedConversionCastOp>(loc, newType, input)
+                .getResult(0));
+      }
+    }
+
+    llvm::SmallVector<Type> resultTypes;
+    for (auto resultType : op.getResultTypes()) {
+      resultTypes.push_back(convertShapedToSignless(resultType));
+    }
```

Comment:
This one should happen automagically if you add the conversion to the type converter

Reply:
Managed to rework with the type converter.